src/System.Management.Automation/engine/CommandCompletion/ExtensibleCompletion.cs

```diff
@@ -207,4 +207,168 @@ public IEnumerable<CompletionResult> CompleteArgument(string commandName, string
+            if (StartWithWordToComplete(text: text) || StartWithWordToComplete(listItemText) || StartWithWordToComplete(toolTip))
+                CompleteWith(text: text, listItemText: listItemText ?? text, toolTip: toolTip ?? text,
+                    resultType: resultType);
+        }
+
+
+        /// <summary>
+        ///     Adds a completion result to the result set if the any string argument starts with <see cref="WordToComplete" />
+        /// </summary>
+        /// <para>The comparison is case insensitive</para>
+        /// <param name="text">the text to be used as the auto completion result</param>
+        /// <param name="listItemText">the text to be displayed in a list</param>
+        /// <param name="toolTip">the text for the tooltip with details to be displayed about the object</param>
+        /// <param name="resultType">the type of completion result</param>
+        protected void CompleteWithIfAnyContainsWordToComplete(string text, string listItemText = null,
+            string toolTip = null, CompletionResultType resultType = CompletionResultType.ParameterValue)
+        {
+            if (ContainsWordToComplete(text) || ContainsWordToComplete(listItemText) || ContainsWordToComplete(toolTip))
+                CompleteWith(text: text, listItemText: listItemText ?? text, toolTip: toolTip ?? text,
+                    resultType: resultType);
+        }
+
+        /// <summary>
+        ///     Adds a completion result to the result set if the text contains <see cref="WordToComplete" />
+        /// </summary>
+        /// <para>The comparison is case insensitive</para>
+        /// <param name="text">the text to be used as the auto completion result</param>
+        /// <param name="listItemText">the text to be displayed in a list</param>
+        /// <param name="toolTip">the text for the tooltip with details to be displayed about the object</param>
+        /// <param name="resultType">the type of completion result</param>
+        protected void CompleteWithIfTextContainsWordToComplete(string text, string listItemText = null,
+            string toolTip = null, CompletionResultType resultType = CompletionResultType.ParameterValue)
+        {
+            if (ContainsWordToComplete(text))
+                CompleteWith(text: text, listItemText: listItemText ?? text, toolTip: toolTip ?? text,
+                    resultType: resultType);
+        }
+
+        private static string QuoteIfSpace(string text)
+            => text.Contains(" ") ? $@"""{text}""" : text;
+
+        /// <summary>
+        ///     Predicate to test if a string starts with <see cref="WordToComplete" />
+        /// </summary>
+        /// <param name="text"></param>
+        /// <returns>true if the text contains <see cref="WordToComplete" />, otherwise false</returns>
+        protected bool StartWithWordToComplete(string text)
+            => text.StartsWith(value: WordToComplete, comparisonType: StringComparison.CurrentCultureIgnoreCase);
```

Comment:
Following on from @lzybkr's [link](https://github.com/lzybkr/TabExpansionPlusPlus/blob/deea6b96fcfb290f98da44104336f27d0be68fed/WindowsExe.ArgumentCompleters.ps1#L16), is `OrginalIgnoreCase` maybe a more correct default? (Says the American that never uses alphabets where they're not equivalent.)